pkg/parser/parser_test.go

```diff
@@ -170,17 +177,10 @@ var _ = Describe("Assert generated customData and cseCmd", func() {
 		if err != nil {
 			log.Printf("Failed to marshal the nbcontractv1 to json: %v", err)
 		}
-		cseCmd, err := parser.Parse(inputJSON)
+		cseCmd, err := Parse(inputJSON)
 		Expect(err).To(BeNil())
 
-		if generateTestData() {
-			if _, err = os.Stat(fmt.Sprintf("./testdata/%s", folder)); os.IsNotExist(err) {
-				e := os.MkdirAll(fmt.Sprintf("./testdata/%s", folder), 0755)
-				Expect(e).To(BeNil())
-			}
-			err = os.WriteFile(fmt.Sprintf("./testdata/%s/generatedCSECommand", folder), []byte(cseCmd), 0644)
-			Expect(err).To(BeNil())
-		}
+		generateTestDataIfRequested(folder, cseCmd)
```

Comment:
@lilypan26  Refactored a bit your function so that I can re-use it in my test. 😊